src/commands/cosmos_mode.rs

```diff
@@ -0,0 +1,133 @@
+//! `Cosmos mode signer` subcommand - example of how to write a subcommand
+
+/// App-local prelude includes `app_reader()`/`app_writer()`/`app_config()`
+/// accessors along with logging macros. Customize as you see fit.
+use crate::prelude::*;
+use abscissa_core::error::BoxError;
+use futures::future;
+use signatory::FsKeyStore;
+
+use crate::application::APP;
+use ethers::{
+    prelude::*,
+    providers::{Http, Provider},
+};
+
+use crate::{
+    collector::{Collector, Poller, Request, Response},
+    config::CellarRebalancerConfig,
+};
+use abscissa_core::{config, Command, FrameworkError, Options, Runnable};
+use std::{convert::TryFrom, path, sync::Arc};
+use tokio::task::JoinHandle;
+use tower::{Service, ServiceBuilder};
+
+/// `Cosmos mode signer` subcommand
+///
+/// The `Options` proc macro generates an option parser based on the struct
+/// definition, and is defined in the `gumdrop` crate. See their documentation
+/// for a more comprehensive example:
+///
+/// <https://docs.rs/gumdrop/>
+#[derive(Command, Debug, Options)]
+pub struct CosmosSignerCmd {
+    /// To whom are we saying hello?
+    #[options(free)]
+    recipient: Vec<String>,
+}
+
+impl CosmosSignerCmd {
+    /// Initialize collector poller (if configured/needed)
+    async fn build_pollers(
+        &self,
+        config: CellarRebalancerConfig,
+    ) -> Vec<Poller<SignerMiddleware<Provider<Http>, Wallet<k256::ecdsa::SigningKey>>>> {
+        let keystore = path::Path::new(&config.keys.keystore);
+        let keystore = FsKeyStore::create_or_open(keystore).expect("Could not open keystore");
+
+        let name = &config
+            .keys
+            .rebalancer_key
+            .parse()
+            .expect("Could not parse name");
+        let key = keystore.load(name).expect("Could not load key");
+
+        let key = key
+            .to_pem()
+            .parse::<k256::elliptic_curve::SecretKey<k256::Secp256k1>>()
+            .expect("Could not parse key");
+
+        let wallet: LocalWallet = Wallet::from(key);
+
+        let eth_host = config.ethereum.rpc.clone();
+
+        let mut pollers = vec![];
+
+        for cellar in config.cellars.clone().into_iter() {
+            let client = Provider::<Http>::try_from(eth_host.clone()).unwrap();
+            let client = SignerMiddleware::new(client, wallet.clone());
+            let client = Arc::new(client);
+            let mongo = config.mongo.clone();
+
+            let name = &config
+            .keys
+            .rebalancer_key;
+            let cosmos_key = config.load_deep_space_key(name.clone());
+
+            let poller = Poller::new(&cellar, client, &mongo, &cosmos_key)
```

Comment:
So I think hen the poller gets constructed an argument should be passed in determining if it's in single signer or cosmos mode